src/Semi.Avalonia/Controls/ToggleSwitch.axaml

```diff
@@ -146,19 +151,99 @@
                 <Setter Property="ContentPresenter.IsVisible" Value="False" />
             </Style>
             <Style Selector="^ /template/ Border#SwitchBackgroundBorder">
-                <Setter Property="Border.Background" Value="{DynamicResource ToggleSwitchContainerCheckedDefaultBackground}" />
+                <Setter Property="Border.Background"
+                        Value="{DynamicResource ToggleSwitchContainerCheckedDefaultBackground}" />
             </Style>
             <Style Selector="^:pointerover /template/ Border#SwitchBackgroundBorder">
-                <Setter Property="Border.Background" Value="{DynamicResource ToggleSwitchContainerCheckedPointerOverBackground}" />
+                <Setter Property="Border.Background"
+                        Value="{DynamicResource ToggleSwitchContainerCheckedPointerOverBackground}" />
             </Style>
             <Style Selector="^:pressed /template/ Border#SwitchBackgroundBorder">
-                <Setter Property="Border.Background" Value="{DynamicResource ToggleSwitchContainerCheckedPressedBackground}" />
+                <Setter Property="Border.Background"
+                        Value="{DynamicResource ToggleSwitchContainerCheckedPressedBackground}" />
             </Style>
             <Style Selector="^:disabled /template/ Border#SwitchBackgroundBorder">
-                <Setter Property="Border.Background" Value="{DynamicResource ToggleSwitchContainerCheckedDisabledBackground}" />
+                <Setter Property="Border.Background"
+                        Value="{DynamicResource ToggleSwitchContainerCheckedDisabledBackground}" />
             </Style>
         </Style>
+    </ControlTheme>
 
+    <ControlTheme x:Key="CustomToggleSwitch" BasedOn="{StaticResource {x:Type ToggleSwitch}}" TargetType="ToggleSwitch">
+        <Setter Property="Template">
+            <ControlTemplate TargetType="ToggleSwitch">
+                <Grid
+                    Background="Transparent"
+                    Cursor="Hand"
+                    ColumnDefinitions="Auto">
+                    <Grid x:Name="PART_SwitchKnob" />
+                    <Grid x:Name="PART_MovingKnobs" />
+                    <ContentPresenter
+                        x:Name="PART_OnContentPresenter"
+                        Margin="{DynamicResource ToggleSwitchOnContentMargin}"
+                        Content="{TemplateBinding OnContent}"
+                        ContentTemplate="{TemplateBinding OnContentTemplate}" />
+                    <ContentPresenter
+                        x:Name="PART_OffContentPresenter"
+                        Margin="{DynamicResource ToggleSwitchOnContentMargin}"
+                        Content="{TemplateBinding OffContent}"
+                        ContentTemplate="{TemplateBinding OffContentTemplate}" />
+                </Grid>
+            </ControlTemplate>
+        </Setter>
```

Comment:
Consider add :pointerover/:pressed style similar to borderless button. 
So:
1. Use Border instead of Grid as container
2. Add :pointerover/:pressed sub style. color can be copied from borderless button.